### PR TITLE
[#9510] Instructor: Course Page/Student List Component: Implement sorting for table headers

### LINKED ISSUE #9510: Instructor: Course Page/Student List Component: Implement sorting for table headers
**Environment**: 
`master` branch

**Description of feature/enhancement**
Allow instructors to sort entries in tables located in the instructor course page and also the student list component.

### PR DESCRIPTION
**Pull request by pbrahmbhatt3**
Fixes #9510 

Added sort functions in instructor-courses-page.components.ts to sort activeCourses, archivedCourses and softDeletedCourses. 
Added buttons on in instructor-courses-page.component.html to sort the course list.

I would like to know more about how would you like to sort the studentList component as it is already sorted by team name.

### CODE REVIEW

**Comment by pbrahmbhatt3** [comment]
> 
> 
> Hi @pbrahmbhatt3, these parts of your pull request do not appear to follow our [contributing guidelines](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr):
> 
>     1. PR Title
>        
>        * Issue Reference (`#<issue-number>`) missing.

I have updated PR's title to include issue number

**Comment by pbrahmbhatt3** [comment]
@tanhengyeow  I have updated the files correctly to fix the issue. However, I am not sure why the test is failing. Could please advice me in what could I correct to solve it! Your help will be much appreciated

**Comment by pbrahmbhatt3** [comment]
I found out that there is an error in feedback-questions.service.ts. 
image

It has nothing to do with the changes I have made. I am not sure how to fix it

File: package.json [hunk 1]
```diff
@@ -48,9 +48,9 @@
   },
   "devDependencies": {
     "@angular-builders/jest": "~7.1.2",
-    "@angular-devkit/build-angular": "~0.11.4",
+    "@angular-devkit/build-angular": "^0.13.7",
```
Comment (RonakLakhotia) [inline package.json]: Revert this unrelated change

File: package.json [hunk 2]
```diff
@@ -48,9 +48,9 @@
...
     "@angular/cli": "~7.1.4",
-    "@angular/compiler-cli": "~7.1.4",
+    "@angular/compiler-cli": "^7.2.11",
```
Comment (RonakLakhotia) [inline package.json]: Revert this too

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.html [hunk 3]
```diff
@@ -121,6 +131,16 @@ <h2>Active Courses</h2>
   <h2 class="text-muted">
     <span class="fa fa-file-archive"></span> Archived Courses
   </h2>
+  <div class="col-8">
+      <div class="float-right">
+        <h5 class="d-inline"><strong> Sort By: </strong></h5>
+        <div class="btn-group" data-toggle="buttons">
+          <button class="btn btn-light" (click)="sortById('archieve')">Course ID</button>
```
Comment (RonakLakhotia) [inline src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.html]: Not sure what you mean by archieve here, do you mean archive?

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts [hunk 4]
```diff
@@ -82,10 +82,10 @@ export class InstructorCoursesPageComponent implements OnInit {
   canRestoreAll: boolean = true;
 
   constructor(private route: ActivatedRoute,
-              private httpRequestService: HttpRequestService,
-              private statusMessageService: StatusMessageService,
-              private courseService: CourseService,
-              private modalService: NgbModal) { }
+    private httpRequestService: HttpRequestService,
```
Comment (RonakLakhotia) [inline src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts]: Why this change in indentation?

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts [hunk 5]
```diff
@@ -208,7 +208,7 @@ export class InstructorCoursesPageComponent implements OnInit {
       }, (resp: ErrorMessageOutput) => {
         this.statusMessageService.showErrorMessage(resp.error.message);
       });
-    }, () => {});
+    }, () => { });
```
Comment (RonakLakhotia) [inline src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts]: Unrelated change

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts [hunk 6]
```diff
@@ -270,7 +305,7 @@ export class InstructorCoursesPageComponent implements OnInit {
         this.statusMessageService.showErrorMessage(resp.error.message);
       });
 
-    }, () => {});
+    }, () => { });
```
Comment (RonakLakhotia) [inline src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts]: Unrelated change

**Comment by pbrahmbhatt3** [comment]
Hi @RonakLakhotia,

Thank you for your suggestions! I have updated my code. I have followed Instructor-home-page component to implement sorting for instructor-courses-page. I added 2 fields called SortBy and instructorCourseSortBy to implement sorting. 

However, the tests fail saying  Received value does not match stored snapshot "InstructorCoursesPageComponent should snap with default fields 1". Do I also have to update the snapshot of this component to include these fields?

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts [hunk 7]
```diff
@@ -223,6 +223,84 @@ export class InstructorCoursesPageComponent implements OnInit {
     }, () => {});
   }
 
+  /**
+   * Sorts the panels of courses in order.
+   */
+  sortPanelsBy(by: string):
+      ((a: ActiveCourse, b: ActiveCourse) => number) {
+    return ((a: ActiveCourse, b: ActiveCourse): number => {
+      let strA: string;
```
Comment (RonakLakhotia) [inline src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts]: These variables name don't seem to be intuitive. Perhaps use a more relatable name?

File: package.json (hunk 1 above)
Comment (RonakLakhotia) [reply package.json]: Also, don't forget to revert these changes

**Comment by xpdavid** [comment]
Close due to inactivity.
